### PR TITLE
Add a header for noindex and nofollow

### PR DESCRIPTION
**Pull request by david4096**
This will hopefully stop most search engines from indexing non-production Dockstore instances

**Description**

During audit it was determined we should limit indexing to non-production Dockstore instances. Though we don't control exactly how indexers behave, this method appears more effective (for Google at least) than robots.txt.

**Review Instructions**

To review this ticket ensure that the non-production deploys have the `X-Robots-Tag noindex, nofollow` header applied using `curl -I qa.dockstore.org`. Following this by a reindex request (or simply waiting), may give us the desired results of removing dev/qa/staging from search results.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-4908

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.

### CODE REVIEW

**Review by svonworl — APPROVED** [review]

File: templates/default.nginx_http.security.conf.template
Comment (coverbeck) [inline templates/default.nginx_http.security.conf.template:32]: This is going to add the header for all responses, including API responses. Ideally we would only add it for HTML responses, although it doesn't hurt. Optional

**Review by coverbeck — APPROVED** [review]
Nice find! Looks like the robots.txt doesn't undo existing indexes, which is why this is needed. But the robots.txt is doing something; if you google "staging dockstore", you'll see that it is (sort of) blocked by robots.txt

Do you want to target this to the hotfix? Your call, just making sure it's OK to wait.

**Review by denis-yuen — APPROVED** [review]
checklist item though

**Review by kathy-t — APPROVED** [review]

File: templates/default.nginx_http.security.conf.template
Comment (david4096) [reply templates/default.nginx_http.security.conf.template:32]: Will leave that to be addressed with https://ucsc-cgl.atlassian.net/browse/DOCK-1754

**Pull request merged**
